### DISCUSSION TITLE
无法登录，尝试了各种协议都不行

### DISCUSSION THREAD

**Discussion by ityaquan** [simple-robot/Discussions#119] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

**依赖**
```
<dependency>
    <groupId>love.forte.simbot.boot</groupId>
    <artifactId>simboot-core-spring-boot-starter</artifactId>
    <version>3.0.0-M6</version>
</dependency>
<dependency>
    <groupId>love.forte.simbot.component</groupId>
    <artifactId>simbot-component-mirai-core</artifactId>
    <version>3.0.0.0-M5</version>
</dependency>
```

**配置文件**
```
{
  "component": "simbot.mirai",
  "code": "qq",
  "passwordInfo": {
    "type": "text",
    "text": "密码"
  },
  "config": {
    "protocol": "协议"
  }
}
```

**ANDROID_PHONE**
`net.mamoe.mirai.network.WrongPasswordException: Error(bot=Bot(329249172), code=45, title=禁止登录, message=你当前使用的QQ版本过低，请前往QQ官网im.qq.com下载最新版QQ后重试。, errorInfo=)`

**ANDROID_PAD**
`net.mamoe.mirai.network.WrongPasswordException: Error(bot=Bot(329249172), code=45, title=禁止登录, message=你当前使用的QQ版本过低，请前往QQ官网im.qq.com下载最新版QQ后重试。, errorInfo=)`

**ANDROID_WATCH**
`net.mamoe.mirai.network.WrongPasswordException: Error(bot=Bot(329249172), code=238, title=禁止登录, message=请使用手Q扫描二维码或短信登录。, errorInfo=)`

**IPAD**
`net.mamoe.mirai.network.WrongPasswordException: Error(bot=Bot(329249172), code=45, title=禁止登录, message=你当前使用的QQ版本过低，请前往QQ官网im.qq.com下载最新版QQ后重试。, errorInfo=)`

**MACOS**
`net.mamoe.mirai.network.WrongPasswordException: Error(bot=Bot(329249172), code=45, title=禁止登录, message=你当前使用的QQ版本过低，请前往QQ官网im.qq.com下载最新版QQ后重试。, errorInfo=)`

**注：更换QQ、删除cache和device.json文件的方式均使用过，还是一样报上面的错**

**Comment by ForteScarlet** [comment]
1. 首先依赖版本太低。参考 [核心库Releases](https://github.com/simple-robot/simpler-robot/releases) 和 [mirai组件库Releases](https://github.com/simple-robot/simbot-component-mirai/releases) 来更新版本
2. 参考官方的汇总说明 https://github.com/mamoe/mirai/issues/2570 ，且如果mirai组件库中mirai版本不够新，可以尝试手动排除并替换 `mirai-core` 的版本
3. 清理 `device.json` 、`cache` 等内容，且更新后注意参考 https://github.com/simple-robot/simbot-component-mirai/pull/120 中的不兼容注意事项
